### DISCUSSION TITLE
Filetypes confusion? markdown.vim and lsp_markdown.vim

### DISCUSSION THREAD

**Discussion by ilvez** [neovim/neovim#27966] [General]
There are 2 markdown filetypes defined: `markdown.vim` and `lsp_markdown.vim`. I discovered this because for default `markdown` filetype, that gets associated with `.md` files by default, the highlighting does not work and for `lsp_markdown` it does.

This leads me to question these questions:
- Why do we need 2 separate filetypes for markdown?
- If we look at `lsp_markdown.vim` then this includes `markdown.vim` as a first thing.  Can't there be some conditional in the `markdown.vim` which would just run the stuff in `lsp_markdown.vim` when it's met?

If I would get some solid feedback on this, I could make it into a PR, but it would be cool to hear some context first. Maybe there's a good reason why it is so.

**Comment by clason** [comment]
The issue is that we do not maintain these syntax files; they are taken verbatim from Vim, who would look poorly on us trying to shove _any_ Neovim-specific modifications into them.

Just ignore the syntax file; it's obsolete after switching to tree-sitter parsing for LSP content and will be removed.
